Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -450,7 +450,7 @@ public class ProgramBuilder {
             for (idx, input) in instr.inputs.enumerated() {
                 neededInputs.append(input)
                 if probability(0.2) && mode != .conservative {
-                    var type = program.types.getType(of: input, at: instr.index)
+                    var type = program.type(of: input, after: instr.index)
```

Comment:
I think here we actually need `program.type(of: input, before: instr.index)` because we are changing the input of `instr`, so we need the types before that instruction executed. Maybe wen can just have `Program.type(of:, before:)` which simply calls the other function with `index - 1`?